go/felix/usagerep/usagerep_test.go

```diff
@@ -60,4 +61,27 @@ var _ = Describe("Usagerep", func() {
 		Expect(q.Get("guid")).To(Equal("baddecaf"))
 		Expect(q.Get("cluster_type")).To(Equal("unknown"))
 	})
+	It("should delay at least 5 minutes", func() {
+		Expect(calculateInitialDelay(0)).To(BeNumerically(">=", 5*time.Minute))
+		Expect(calculateInitialDelay(1)).To(BeNumerically(">=", 5*time.Minute))
+		Expect(calculateInitialDelay(1000)).To(BeNumerically(">=", 5*time.Minute))
+	})
+	It("should have a random component", func() {
+		Expect(calculateInitialDelay(1000000)).ToNot(Equal(calculateInitialDelay(1000000)))
+	})
+	It("should have an average close to expected value", func() {
+		var total time.Duration
+		for i := int64(0); i < 1000000; i++ {
+			total += calculateInitialDelay(60)
+			if i > 100 {
+				average := time.Duration(int64(total) / (i + 1))
+				if average > (5*time.Minute+20*time.Second) &&
```

Comment:
I think it's worth commenting here that the average jitter is 0.5s per host, and so 30s for 60 hosts - and hence the test for the average being between 20s and 40s.